internal/relation/endpoint.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package relation
```

Comment:
This new file is the primary change.

Reply:
I will add a package godoc. However I will not be adding all of the missing godoc files for packages touched by this refactoring. Those can be added when substantive changes are made to them.